src/Symfony/Component/DependencyInjection/Compiler/RegisterServiceSubscribersPass.php

```diff
@@ -72,7 +72,7 @@ protected function processValue($value, bool $isRoot = false)
         $subscriberMap = [];
 
         foreach ($class::getSubscribedServices() as $key => $type) {
-            if (!\is_string($type) || !preg_match('/^\??[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)*+$/', $type)) {
```

Comment:
Removing this is what allowed union types - not sure if we want to modify this instead?

Reply:
```php
if (!\is_string($type) || !preg_match('/(?(DEFINE)(?<cn>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+))(?(DEFINE)(?<fqcn>(?&cn)(?:\\\\(?&cn))*+))^(?&fqcn)(?:(?:\|(?&fqcn))*+|(?:&(?&fqcn))*+)$/x', $type)) {
```

Blows up the tests - is this what you mean?

Reply:
yep, without the `x` if you don't need it (it allows adding whitespace/comments into the regex)

Reply:
Fixed, didn't account for the possible `?` prefix.